### PR TITLE
build: add support for multiple Github Templates

### PR DESCRIPTION
**Pull request by Cobalt0s**
@RajatPawar please have a look

# How to add templates
Create `<filename>.md` under `.github/PULL_REQUEST_TEMPLATE`.
Ensure that filename has no spaces. It is used as query parameter.

# Script
Bash script implies Pull Request URL based on current branch.
Appends for every template a respective query parameter. See [template qp](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request).
image

# Reference
[Github docs.](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)

# IMPORTANT
In order to see the changes in action `.github/PULL_REQUEST_TEMPLATE` must be present on default branch first!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/example.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Ticket Number
+
+Create other templates and reference them with query params attached to URL creating PR.
```
Comment (laurenzlong) [inline .github/PULL_REQUEST_TEMPLATE/example.md:3]: This seems to be just be an example file, can we remove it from this PR so it's not checked in as a real template?

File: .github/PULL_REQUEST_TEMPLATE/new_connector.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+## Checklist
+- [ ] Ran Linter
+- [ ] Catalog tests passing
+- [ ] Created PR from non-main branch
```
Comment (laurenzlong) [inline .github/PULL_REQUEST_TEMPLATE/new_connector.md:4]: This third item is probably not needed, Github wouldn't have let you create a PR from a main branch.

File: .github/PULL_REQUEST_TEMPLATE/new_connector.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## Catalog variables
+<Please mention any variables you had to use for the provider's info>
```
Comment (laurenzlong) [inline .github/PULL_REQUEST_TEMPLATE/new_connector.md:7]:
```suggestion
<Does the provider's info include any template variables such as `{{workspace}}`?>
```

File: Makefile [hunk 4]
```diff
@@ -26,3 +26,10 @@ format: fix
 .PHONY: test
 test:
 	go test -v ./...
+
+# Creates PR URLs for each template
+# Click on one of them or manually add ?template=<file.md> to the URL if you are creating via Github website
```
Comment (laurenzlong) [inline Makefile:31]:
```suggestion
# Click on one of them or manually add ?template=<file.md> to the URL if you are creating a PR via the Github website
```

File: Makefile [hunk 5]
```diff
@@ -26,3 +26,10 @@ format: fix
...
+# Templates: Under github/PULL_REQUEST_TEMPLATE directory you can add more templates and script handles suffixing automatically
```
Comment (laurenzlong) [inline Makefile:32]:
```suggestion
# Templates: Under github/PULL_REQUEST_TEMPLATE directory you can add more templates
```

File: scripts/bash/pr_options.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+#
+# Referenced from Makefile
+#
+pr_template() {
+	SOURCE_BRANCH=$(git rev-parse --abbrev-ref HEAD); # current branch from which PR is created
```
Comment (laurenzlong) [inline scripts/bash/pr_options.sh:7]:
```suggestion
  SOURCE_BRANCH=$(git rev-parse --abbrev-ref HEAD); # current branch from which PR is created
```

**Review by laurenzlong — APPROVED** [review]
This is a really great idea, thank you so much @Cobalt0s !

File: .github/PULL_REQUEST_TEMPLATE/new_connector.md (hunk 2 above)
Comment (RajatPawar) [reply .github/PULL_REQUEST_TEMPLATE/new_connector.md]: This was for forks, but is irrelevant now

**Pull request merged**